beacon-chain/blockchain/core.go

```diff
@@ -194,3 +199,57 @@ func (b *BeaconChain) getAttestersProposer(seed common.Hash) ([]int, int, error)
 	}
 	return indices[:int(attesterCount)], indices[len(indices)-1], nil
 }
+
+func (b *BeaconChain) computeValidatorRewardsAndPenalties() error {
+	activeValidatorSet := b.state.CrystallizedState.ActiveValidators
+	attesterDeposits := b.state.ActiveState.TotalAttesterDeposits
+	totalDeposit := b.state.CrystallizedState.TotalDeposits
+
+	transition := b.IsthereAnEpochTransition()
+	if !transition {
+		return errors.New("invalid slot for epoch transition")
+	}
+
+	if attesterDeposits*3 >= uint64(totalDeposit)*2 {
```

Comment:
magic numbers like these should be put in vars with a comment explaining why we're doing attesterDeposits*3

Reply:
+1

Reply:
Sure thing will add a comment :+1: